Cython/Compiler/Nodes.py

```diff
@@ -901,8 +901,7 @@ def hdr_cname(self):
 
 
     def analyse(self, env, nonempty=0, is_self_arg=False):
-        if is_self_arg:
-            self.base_type.is_self_arg = self.is_self_arg = True
+        self.base_type.is_self_arg = self.is_self_arg = is_self_arg
```

Comment:
Can we keep the `if` test? Given the default class attributes, it doesn't seem necessary to always set it (and that avoids an unnecessary attribute in the instance dicts).